### PR TITLE
[10384] Adding Zero Copy feature to the Latency performance tests

### PR DESCRIPTION
**Pull request by MiguelBarro**
This PR should be merged after PR #1706, #1730 and #1705.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3726/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/903/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3279/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3750/)

File: test/performance/latency/LatencyTestPublisher.cpp [hunk 1]
```diff
@@ -781,8 +781,8 @@ bool LatencyTestPublisher::test(
 
                 if (!loaned)
                 {
-                    logError(LatencyTest, "Problem on Publisher test data with loan");
```
Comment (IkerLuengo) [inline test/performance/latency/LatencyTestPublisher.cpp]: Unless this happens often (and therefore, can be seen as 'normal'), I would keep the Error level log, or at least, Warning level (although not terminating the execution), as it alerts about something smelly on the test.

The same goes for the rest of logs modified in this commit.

File: test/performance/latency/main_LatencyTest.cpp [hunk 2]
```diff
@@ -470,6 +470,18 @@ int main(
         }
     }
 
+    // Check parameters validity
+    if (use_security && test_agent == TestAgent::BOTH)
+    {
+        logError(LatencyTest, "Intra-process delivery NOT supported with security");
+        return 1;
+    }
```
Comment (IkerLuengo) [inline test/performance/latency/main_LatencyTest.cpp:478]: payload or RTPS encryption is not supported with DataSharing either. Should we check it here too?

**Review by IkerLuengo — CHANGES_REQUESTED** [review]
I know we have discussed it already, and we all though it was a good solution, but how fair is to discard the echo bounce time in the measurements? Isn't that part of the 'consequences' of using, for example, the loan API? If having to wait for a loan makes our overall latency worse, should we assume it on the measurements?

File: test/performance/latency/LatencyTestPublisher.cpp (hunk 1 above)
Comment (MiguelBarro) [reply test/performance/latency/LatencyTestPublisher.cpp]: With the larger payload pool and the several trials strategy, these never log. Let's promote them to logError 👍

File: test/performance/latency/main_LatencyTest.cpp (hunk 2 above)
Comment (MiguelBarro) [reply test/performance/latency/main_LatencyTest.cpp:478]: 👍

**Review by IkerLuengo — DISMISSED** [review]
LGTM

**Review by IkerLuengo — DISMISSED** [review]
LGTM

**Review by EduPonz — CHANGES_REQUESTED** [review]
For the not dynamic case, the CSV and STDOUT tables show and incorrect size for bytes larger than what the user requested (68 instead of 64). I have not tested with dynamic data though

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3768/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/939/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3318/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3794/)

**Review by IkerLuengo — APPROVED** [review]
:+1:

**Pull request merged**